### ISSUE TITLE
Trakt.tv

### ISSUE THREAD

**Issue by michou83** [boxeehacks/boxeehack#191]
Trakt.tv is a site that allows you track your TV shows / episodes and Movies, by scrobbling what you are watching on your boxee box in real time. Currently this can be achieved with certain limitations with a Linux server or a NAS running Linux with python 2.7.2 installed, and a little script called trakt-for-boxee. 

If it could be run at startup straight on the Boxee Box, more features could be added. The Boxee Box seems to be running python 2.4, but if python 2.7.2 can be installed trakt-for-boxee could probably be moved as is to the boxee box. 

Few features that might be good: 
1. A front end in the Boxee+ settings, for username and password  for trakt.tv.
2. When marking an episode or movie as watched on the boxee box it would update your trakt.tv database simultaneously.
3. Allow you to rate an episode or movie and those details would be updated to trakt.tv

I'm sure this will be low on the list of priorities on the todo list, but I thought I would mention it anyway.

**Comment by war59312** [comment]
Well if this gets added I will 100% start running Boxee+. I love trakt.tv (of course I am VIP).

I'm sold if this gets done. Willing to pay!!!!

**Comment by diff3** [comment]
I checked on trakt webpage, and I found a php script for boxee 0.9. I got php running on boxee so I guess if you want trakt you got it. I also got cron working. 

http://trakt.tv/downloads/boxee (look for How to use the trakt PHP script for Boxee 0.9+ )
boxee https://github.com/rudf0rd/trakt-standalone

you can find my files on https://github.com/boxeehacks/boxeehack/issues/217

**Comment by diff3** [comment]
First try running trakt failed, need to recompile php for bb

**Comment by diff3** [comment]
Recompiled php with curl support and added some new libs. Got trakt running with trakt-standalone script for boxee 0.9. got to do some more testing, but at least I got connected with both boxee box and trakt.tv with the php script

**Comment by BjornB2** [comment]
Full Trakt.tv integration is not only about scrobbling, but also about setting movies/tv episodes to watched when they are watched on another device. 
The above is of course a way to get trakt.tv working, but maybe not very user-friendly. It would be great if in settings there would be an option to login with my trakt account and that 2 way sync would work.

Now I have a python script running on my NAS, but that's only one way.

**Comment by michou83** [comment]
I agree with BjornB2, as I mentioned in my first post, I believe a user interface would be needed ultimately in Settings --> Boxee+ .With options for:  
1. Enable 2 way syncing with trakt.tv, 
2. Enter a Username and Password for your trakt.tv account. 
3. Multiple options to choose what to sync/scrobble (Movies, TV, Marked Watched/Unwatched, Star Ratings)

The main stumbling blocks I believe would be: 
1. Can the Boxee Box own database be updated from trakt.tv database, which would mark already watched episodes/movies as Watched.
2. When marking an episode as watched/unwatched on the Boxee Box, how easily can that episode/movie information be extracted and then sent to the trakt.tv database.
